Storage/SQL/SQLiteHistory.swift

```diff
@@ -541,6 +559,34 @@ extension SQLiteHistory: BrowserHistory {
             "LIMIT \(historyLimit)",
         ].joined(separator: " ")
 
+        if usePregeneratedTempTable && AppConstants.BuildChannel != .release {
+            db.withConnection { connection -> Void in
+                do {
+                    try connection.executeChange("select * FROM sqlite_master WHERE name='\(AwesomebarBookmarksTempTable)'")
```

Comment:
What is this supposed to do?

Reply:
In debug, verify that the temp table has been generated. If the table doesn't exist (i.e. the select fails), log an error.

`getSitesByFrecencyWithHistoryLimit(..., usePregeneratedTempTable: Bool)` is tightly coupled with `createTempTablesForFrecencyQuery()` in that if `usePregeneratedTempTable: true` is used then `createTempTablesForFrecencyQuery()` *must* have preceded this call.

*ASIDE: I don't like this coupling, but seems acceptable in this case given it is specialized to this use and I don't have a better idea.*

Reply:
Hence my suggestion of using the type system to ensure that the appropriate preconditions must have been met in order to call that method.

Specifically in the implementation above, though:

1. I'd prefer this to be phrased more explicitly:

```
// In pre-release channels, verify that the bookmarks temp table exists.
let shouldAssertTableExists = usePregeneratedTempTable && AppConstants.BuildChannel != .release
```

2. I don't believe that query will err if the table is missing, so I'm not sure this is doing the validation you expect.

Reply:
And yeah, it just returns 0 rows safely. Oopsy.

Reply:
Agreed BTW to your suggested syntax change for greater clarity.